File: algorithms/python3/moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        writePointer = 0
        for readPointer in range(len(nums)):
            if (readPointer == len(nums) or nums[readPointer] != 0):
                nums[writePointer] = nums[readPointer]
                if (readPointer != writePointer):
                    nums[readPointer] = 0
                writePointer += 1

File: algorithms/c/moveZeroes.c
void moveZeroes(int* nums, int numsSize){
    int writePointer = 0;
    for (int readPointer = 0; readPointer < numsSize; readPointer++) {
        if (nums[readPointer] != 0) {
            nums[writePointer] = nums[readPointer];
            if (writePointer != readPointer) {
                nums[readPointer] = 0;
            }
            writePointer++;
        }
    }
    
}
